doc/deployment-schema-1.0.json

```diff
@@ -257,7 +212,8 @@
                                             "type": "string",
                                             "examples": [
                                                 "FROM /* INTO $upstream"
-                                            ]
+                                            ],
+                                            "pattern": "^(?i)FROM\\s\/.*?(\\s|\\sWHERE\\s.*?\\s)INTO\\s(\\$upstream|BrokeredEndpoint\\(.*?\\))$"
```

Comment:
Seems the pattern is not valid. I tried to use the schema to validate deployment manifest. And Got error:  SyntaxError: Invalid regular expression: /^(?i)FROM\s/.*?(\s|\sWHERE\s.*?\s)INTO\s(\$upstream|BrokeredEndpoint\(.*?\))$/: Invalid group

Reply:
I think writing a valid regex expression for the routing rules is going to be challenging/impossible. We may want to figure out a way to do this outside of the schema or simply add more examples.

I can speak on this pretty confidently because I wrote the original routing rule parser. This is in Antlr which is much more sophisticated than regex. Because of the set of characters IoT Hub allows in properties, it's very difficult to create a valid parser.

Reply:
I reverted the pattern to simply check at least one character